compiler/damlc/tests/daml-test-files/DamlDocHoogle.EXPECTED.hoogle

```diff
@@ -0,0 +1,57 @@
+
+module DamlDocHoogle
+
+-- | View docs
+
+data View
+
+
+View :: View
+
+
+getController :: I -> Party
+
+-- | T docs
+
+data T
+
+-- | T docs
+
+T :: Template T => Party -> T
+
+-- | TCall docs
+
+data TCall
+
+-- | TCall docs
+
+TCall :: Int -> TCall
+
+[i] :: TCall -> Int
+
+-- | TCall docs
+
+TCall_Choice :: Choice T => Int -> Update ()
```

Comment:
Right right, thats the purpose of this constructor though, its to easily show the arguments you need and the return type.
Its a convenience to avoid having to then look for the TCall data type.
Keep in mind, this is solely for quick lookup, the intent is that you click it to get the real information. (and we should communicate that)

Reply:
I do agree there, and its a shame we don't have more power over hoogle here.
I'll note we're using `::` over `:` here which might give us some leeway ;)

Both results would show up, but not always near each other, which is why I leaned towards the shared prefix.
I'm unsure, when we lean towards exercise_TCall, we're closer to the real thing, which might end up being even more misleading, as these two aren't that far different
```hs
exercise_TCall cid (TCall 1)
```
```hs
exercise cid (TCall 1)
```
Maybe thats close enough though.
Looking up `exercise` with this approach would be pain though, as you'd get all the specialised ones with the real one

Reply:
This whole thing is quite painful, because at the end of the day, its a hack. Hoogle was absolutely not made for this

Reply:
> Looking up `exercise` with this approach would be pain though

ouch, that's true

> This whole thing is quite painful, because at the end of the day, its a hack. Hoogle was absolutely not made for this

yeah :( it's something to keep in mind for later

Reply:
Heres a decent middle ground
![image](https://github.com/digital-asset/daml/assets/123961959/602eb4fe-fe23-40e4-82c1-99a3523565fb)
We get a fair bit of liberty with the constraints, we could literally write `Choice For T Returning ()` if we wanted
`Choice T ()` could also work, or even the actual HasExercise definition, giving
`data HasExercise T Call () => Call`, ofc along side its normal constructor